lib/gitarro/backend.rb

```diff
@@ -63,105 +63,118 @@ def initialize(option = nil)
     @gbexec = TestExecutor.new(@options)
   end
 
-  # public method for get prs opens
-  # given a repo
-  def open_prs
-    prs = @client.pull_requests(@repo, state: 'open')
-    puts 'no Pull request OPEN on the REPO!' unless prs.any?
+  # Check if a PR list is empty
+  # If it is and we do want to consider the update time, return true
+  # Otherwise return false
+  def pr_list_empty?(prs)
+    prs.empty? && @options[:changed_since] >= 0 ? true : false
+  end
+
+  # public method for get prs opened and matching the changed_since
+  # condition
+  def open_newer_prs
+    prs = []
```

Comment:
Ntipick: we can refactor this with a beatiful ruby enumerable class.

```console
prs = @client.pull_requests(@repo, state: 'open').select do 
  pr_last_update_less_than(pr, @options[:changed_since])
end
```

Reply:
```console
$ irb
irb(main):001:0> [1,2,3].select {|n| n > 4 }
=> []
```
http://ruby-doc.org/core-2.2.2/Enumerable.html#method-i-select

Reply:
Done.